tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -421,3 +424,124 @@ def prepare_ion_recomb_coefficients_nlte_ion(
             total_coll_ion_coefficients,
             total_coll_recomb_coefficients,
         )
+
+    def jacobian_matrix(
+        self,
+        populations,
+        rate_matrix,
+        rate_matrix_index,
+        total_rad_recomb_coefficients,
+        total_coll_ion_coefficients,
+        total_coll_recomb_coefficients,
```

Comment:
Since these coefficients are being passed through a few functions, they might be better as class properties if that would work with the calculations.

Reply:
I think it should, those don't change during the calculation.

Reply:
That would help simplify the function inputs/outputs then, so I recommend it

Reply:
I will change this to take as argument the shell number (id) and get the rates inside functions using that.

Reply:
I agree with @andrewfullard that it might make sense to make the coefficients attributes of the class.

Reply:
Will be dealt with in the future (see #2163 and #2164).